File: Ceresis/App/DTO/workprice.dto.ts
import { RequestBase, ResponseBase } from "./base";

export class WorkPrice {
    id: number;
    name: string;
    unity: string;
    exactPrice: boolean;
    contactPrice: boolean;
    price: number;
}

export class RequestGetWorkprice extends RequestBase {

}

export class ResponseGetWorkprice extends ResponseBase {
    data: WorkPrice[];
}

export class RequestAddWorkprice {
    name: string;
    unity: string;
    exactPrice: boolean;
    contactPrice: boolean;
    price: number;
}

export class ResponseAddWorkprice extends ResponseBase {

}

export class RequestEditWorkprice {
    id: number;
    name: string;
    unity: string;
    exactPrice: boolean;
    contactPrice: boolean;
    price: number;
}

export class ResponseEditWorkprice extends ResponseBase {
    
}

export class RequestDeleteWorkprice {
    id: number;
}

export class ResponseDeleteWorkprice extends ResponseBase {
    
}

File: Ceresis.Data.Core/Model/WorkpriceDTO.cs
﻿using System;
using System.Collections.Generic;
using System.Text;

namespace Ceresis.Data.Core.Model
{
    public class WorkpriceDTO
    {
        public int Id { get; set; }

        public string Name { get; set; }

        public string Unity { get; set; }

        public bool ExactPrice { get; set; }

        public bool ContactPrice { get; set; }

        public double? Price { get; set; }
    }
}
